db/R/dbfiles.R

```diff
@@ -65,12 +66,53 @@ dbfile.input.insert <- function(in.path, in.prefix, siteid, startdate, enddate,
     db.query(cmd, con)
     # return input id
     inputid <- db.query(paste0("SELECT id FROM inputs WHERE site_id=", siteid, " AND format_id=", formatid, " AND start_date='", startdate, "' AND end_date='", enddate, "'" , parent, ";"), con)[['id']]
+  } else {
+    if(nrow(existing.input) > 1) {
+      print(existing.input)
+      logger.warn("Multiple existing inputs found. Using last.")
+      existing.input <- existing.input[nrow(existing.input),]
+    }
+    
+    # Convert dates to Date objects and strip all time zones (DB values are timezone-free)
+    startdate <- force_tz(as_date(startdate), 'GMT')
+    enddate <- force_tz(as_date(enddate), 'GMT')
+    existing.input$start_date <- force_tz(as_date(existing.input$start_date), 'GMT')
+    existing.input$end_date <- force_tz(as_date(existing.input$end_date), 'GMT')
+    
+    if(existing.input$start_date != startdate || existing.input$end_date != enddate) {
+      print(existing.input, digits=10)
+      logger.error(paste0(
```

Comment:
I second that emotion. Isn't this something that can be hard-coded into the PSQL schema? Then there's an extra level of security. Just a thought (brought to mind while I was working on this, and also fighting with BETY to let me delete erroneous records—she clearly knows things like you can't orphan a record with `parent_id`, etc.)